File: frontend/Assets/Scripts/DataTransfer/Client/AuthRequestCmd.cs
using System.IO;
using Attributes;


namespace DataTransfer.Client {

    [DTO(DTOCode.AuthRequest)]
    public class AuthRequestCmd : ClientCommand {

        public int Id { get; private set; }


        public AuthRequestCmd (int id) {
            Id = id;
        }


        public override void WriteMembers (BinaryWriter writer) {
            writer.Write(Id);
        }

    }

}

File: backend/src/main/java/dto/client/AuthRequestCmd.java
package dto.client;

import dto.DTOCode;
import dto.DTOs;
import io.netty.buffer.ByteBuf;
import players.Player;
import players.Players;


@DTOCode(DTOs.AUTH_REQUEST)
public class AuthRequestCmd extends ClientCommand {

    public int id;


    @Override
    public void execute () {
        player = Players.get(id);
        if (player != null) player.disconnect();
    }


    @Override
    public void readMembers (ByteBuf buffer) throws Exception {
        id = buffer.readInt();
    }
}
